src/components/Api.js

```diff
@@ -1,79 +1,80 @@
 export default class Api {
-  constructor({ baseUrl, headers }) {
-    this._baseUrl = baseUrl;
-    this._headers = headers;
-  }
-
-  _handleResponce(res) {
-    if (res.ok) {
-      return res.json();
-    } else {
-      Promise.reject(`Error: ${res.status}`);
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+    }
+  
+    _handleResponce(res) {
+      if (res.ok) {
+        return res.json();
+      } else {
+        Promise.reject(`Error: ${res.status}`);
+      }
+    }
+  
+    getUserInfo() {
+      return fetch(`${this._baseUrl}/users/me`, {
+        headers: this._headers,
+      }).then(this._handleResponce);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleResponce)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.